tests/cli/test_utils.py

```diff
@@ -339,3 +342,37 @@ def test_local_remote_port_param_type_invalid(arg: str) -> None:
     param = LocalRemotePortParamType()
     with pytest.raises(click.BadParameter, match=".* is not a valid port combination"):
         param.convert(arg, None, None)
+
+
+class TestDelayedReleaseWarning:
+    date_to_patch = f"{_need_to_warn_after_delay.__module__}.date"
```

Comment:
better to use the package [freezegun](https://pypi.org/project/pytest-freezegun/), but it requires an additional test dependency:
```
@pytest.mark.freeze_time('2017-05-21')
def test_current_date(current_date):
    pass
```

Reply:
freezegun has own limitations: it doesn't shift the process time but only some Python APIs.
Thus tested code becomes desynchronized easily and quickly.

If you really want to check `datetime` module operations correctness you can pass `now` into `_need_to_warn_after_delay()` helper. 
Quite pointless work though.

Reply:
if it's pointless, should we remove the whole bunch of tests? indeed the logic is very simple there

Reply:
Please remove.
If we change these simple lines patching mocked tests can be tedious.
A chance to make an error in `_need_to_warn_after_delay()` is neglible.